backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,220 @@
+        key_prefix: str,
+        new_dataset_version_id: DatasetVersionId,
+        metadata_update_dict: Dict[str, str],
+    ):
+        h5ad_filename = self.download_from_source_uri(
+            source_uri=h5ad_s3_uri,
+            local_path=CorporaConstants.LABELED_H5AD_ARTIFACT_FILENAME,
+        )
+
+        adata = scanpy.read_h5ad(h5ad_filename)
+        metadata = original_dataset_version.metadata
+        for key, val in metadata_update_dict.items():
+            adata.uns[key] = val
+            if hasattr(metadata, key):
+                setattr(metadata, key, val)
+        adata.write(h5ad_filename)
+        self.business_logic.set_dataset_metadata(new_dataset_version_id, metadata)
+
+        self.create_artifact(
+            h5ad_filename,
+            DatasetArtifactType.H5AD,
+            key_prefix,
+            new_dataset_version_id,
+            self.artifact_bucket,
+            DatasetStatusKey.H5AD,
+            datasets_bucket=self.datasets_bucket,
+        )
+        self.update_processing_status(
+            new_dataset_version_id, DatasetStatusKey.VALIDATION, DatasetValidationStatus.VALID
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.H5AD, DatasetConversionStatus.CONVERTED)
+
+    def update_rds(
+        self,
+        rds_s3_uri: str,
+        key_prefix: str,
+        new_dataset_version_id: DatasetVersionId,
+        metadata_update_dict: Dict[str, str],
+    ):
+        seurat_filename = self.download_from_source_uri(
+            source_uri=rds_s3_uri,
+            local_path=CorporaConstants.LABELED_H5AD_ARTIFACT_FILENAME,
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.CONVERTING)
+
+        base = importr("base")
+        seurat = importr("SeuratObject")
+
+        rds_object = base.readRDS(seurat_filename)
+
+        for key, val in metadata_update_dict.items():
+            seurat_metadata = seurat.Misc(object=rds_object)
+            if seurat_metadata.rx2[key]:
+                seurat_metadata[seurat_metadata.names.index(key)] = StrVector(list(val))
+
+        base.saveRDS(rds_object, file=seurat_filename)
+
+        self.create_artifact(
+            seurat_filename,
+            DatasetArtifactType.RDS,
+            key_prefix,
+            new_dataset_version_id,
+            self.artifact_bucket,
+            DatasetStatusKey.RDS,
+            datasets_bucket=self.datasets_bucket,
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.CONVERTED)
+
+    def update_cxg(
+        self,
+        cxg_s3_uri: str,
+        key_prefix: str,
+        dataset_version_id: DatasetVersionId,
+        metadata_update_dict: Dict[str, str],
+    ):
+        new_cxg_dir = f"s3://{self.cellxgene_bucket}/{key_prefix}.cxg/"
+        self.s3_provider.upload_directory(cxg_s3_uri, new_cxg_dir)
+        ctx = tiledb.Ctx(
```

Comment:
I think I'll factor this out somewhere or at least add a comment. I just copied this from the cxg conversion code